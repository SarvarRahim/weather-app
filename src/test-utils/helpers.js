import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
