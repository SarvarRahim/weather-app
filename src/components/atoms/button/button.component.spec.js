import 'jest-styled-components';
import '@testing-library/jest-dom';
import { customRender } from '@test-utils/helpers';
const { screen } = require('@testing-library/react');

const { Button } = require('./button.component');

const TEST_PROPS = {
  children: 'Some text',
  variant: 'primary',
  size: 'small',
  iconName: 'arrow',
  iconColor: '#fff'
};

const setup = ({ children, variant, size, iconName, iconColor }) => {
  return customRender(
    <Button
      variant={variant}
      size={size}
      iconName={iconName}
      iconColor={iconColor}
      data-testid='button'
      buttonType='button'
    >
      {children}
    </Button>
  );
};

test('Render Button component', () => {
  const { asFragment } = setup(TEST_PROPS);
  expect(asFragment()).toMatchSnapshot();
});

test('Should have children text', () => {
  setup(TEST_PROPS);
  const element = screen.getByTestId('button');
  // eslint-disable-next-line
  expect(element).toHaveTextContent('Some text');
});
