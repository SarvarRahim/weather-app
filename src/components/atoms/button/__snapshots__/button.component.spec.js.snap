// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button component 1`] = `
<DocumentFragment>
  .c1 {
  width: 2rem;
  height: 2rem;
  fill: #000000;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
}

.c1[disabled] {
  opacity: 0.2;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.8rem;
  column-gap: 0.8rem;
  border: none;
  background-color: transparent;
  padding: 0.2rem;
  font-family: ManropeSemiBold;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  padding: 0.6rem 1.8rem;
  border-radius: 10rem;
  font-size: 1.2rem;
  line-height: 2rem;
  background-color: #ba8d5b;
  color: #ffffff;
}

.c0 > svg {
  width: 1.1rem;
  height: 1.1rem;
  fill: #fff;
  stroke: none;
}

.c0[disabled] {
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 0.6;
}

.c0[disabled] > svg {
  opacity: 0.3;
}

.c0 > svg {
  width: 1rem;
  height: 1rem;
}

.c0:hover > svg {
  stroke: none;
}

.c0 > svg {
  fill: #ffffff;
  stroke: none;
}

.c0:hover {
  background-color: #ccad8a;
}

.c0:active {
  background-color: #a77946;
  color: #ffffff;
}

.c0:active > svg {
  fill: #ffffff;
  stroke: none;
}

<button
    class="c0"
    data-testid="button"
    type="button"
  >
    <span>
      Some text
    </span>
    <svg
      class="c1"
    >
      <use
        xlink:href="sprite.svg#arrow"
      />
    </svg>
  </button>
</DocumentFragment>
`;
